homeassistant/components/sensor/yr.py

```diff
@@ -91,11 +91,10 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     async_add_devices(dev)
 
     weather = YrData(hass, coordinates, forecast, dev)
-    # Update weather on the hour, spread seconds
-    async_track_utc_time_change(
-        hass, weather.async_update, minute=randrange(1, 10),
-        second=randrange(0, 59))
-    yield from weather.async_update()
+    async_track_utc_time_change(hass, weather.updating_devices, minute=31)
+    # wait 1 minute after start up before we fetch data,
+    # to avoid several restarts to fetch new data.
+    async_call_later(hass, 60, weather.fetching_data)
```

Comment:
Do we want this?
It will reduce the load on their servers, but the user experience will be lower?

Reply:
This also makes the tests to fail

Reply:
You can also just call it directly. The impact on their servers should be minimal as this is a one time call

Reply:
how about using the last value in recorder ? (like input_*)